consensus/src/round_manager.rs

```diff
@@ -353,23 +360,80 @@ impl RoundManager {
             .proposer_election
             .is_valid_proposer(self.proposal_generator.author(), new_round_event.round)
         {
-            self.log_collected_vote_stats(&new_round_event);
-            self.round_state.setup_leader_timeout();
-            let proposal_msg = self.generate_proposal(new_round_event).await?;
-            #[cfg(feature = "failpoints")]
-            {
-                if self.check_whether_to_inject_reconfiguration_error() {
-                    self.attempt_to_inject_reconfiguration_error(&proposal_msg)
-                        .await?;
+            let epoch_state = self.epoch_state.clone();
+            let round_state = self.round_state.clone();
+            let network = self.network.clone();
+            let sync_info = self.block_store.sync_info();
+            let proposal_generator = self.proposal_generator.clone();
+            let safety_rules = self.safety_rules.clone();
+            let proposer_election = self.proposer_election.clone();
+            tokio::spawn(async move {
+                if let Err(e) = Self::generate_and_send_proposal(
+                    epoch_state,
+                    new_round_event,
+                    round_state,
+                    network,
+                    sync_info,
+                    proposal_generator,
+                    safety_rules,
+                    proposer_election,
+                )
+                .await
+                {
+                    warn!("Error generating and sending proposal: {}", e);
                 }
+            });
+        }
+        Ok(())
+    }
+
+    async fn generate_and_send_proposal(
+        epoch_state: Arc<EpochState>,
+        new_round_event: NewRoundEvent,
+        round_state: Arc<GuardedRoundState>,
+        network: Arc<NetworkSender>,
+        sync_info: SyncInfo,
+        proposal_generator: Arc<ProposalGenerator>,
+        safety_rules: Arc<Mutex<MetricsSafetyRules>>,
+        proposer_election: Arc<dyn ProposerElection + Send + Sync>,
+    ) -> anyhow::Result<()> {
+        let epoch = epoch_state.epoch;
+        Self::log_collected_vote_stats(epoch_state, &new_round_event);
+        round_state.setup_leader_timeout();
```

Comment:
I think we can get rid of this too, it's not useful anymore after all to all broadcast

Reply:
Then, we don't need guarded round state for sure.